cmd/agent/app/agent.go

```diff
@@ -60,12 +62,27 @@ func NewAgentCommand(ctx context.Context) *cobra.Command {
 		},
 	}
 
-	opts.AddFlags(cmd.Flags())
+	opts.AddFlags(cmd.Flags(), KnownControllers())
 	cmd.AddCommand(sharedcommand.NewCmdVersion(os.Stdout, "karmada-agent"))
 	cmd.Flags().AddGoFlagSet(flag.CommandLine)
 	return cmd
 }
 
+// NewControllerInitializers is a public map of named controller groups (you can start more than one in an init func)
+func NewControllerInitializers() map[string]controllerscontext.InitFunc {
```

Comment:
Well, why not make the Initializers as a global variable? It seems we don't need to modify this struct. I prefer this kind of code.

```go
var (
	controllers = make(map[string]controllerscontext.InitFunc)
)

func init() {
	controllers["clusterStatus"] = startClusterStatusController
	controllers["execution"] = startExecutionController
	controllers["workStatus"] = startWorkStatusController
	controllers["serviceExport"] = startServiceExportController
}

type map[string]controllerscontext.InitFunc ControllerInitializers

func (c *ControllerInitializers) ControllerNames() []string {
    return sets.StringKeySet(c).List()
}

func (c *ControllerInitializers) StartControllers(ctx Context) error {
    ......
}
```

Reply:
> Well, why not make the Initializers as a global variable?

Similar to what I mentioned before: https://github.com/karmada-io/karmada/pull/1093#discussion_r767553986, and I'm just following the way the controller-manager, I will update them all.

Reply:
Cool, would you update in this PR or next?

Reply:
Just in this PR, I will update it soon.

Reply:
Updated, PTAL @Garrybest